app/controllers/content_items_controller.rb

```diff
@@ -1,14 +1,22 @@
 class ContentItemsController < ApplicationController
   before_filter :parse_json_request, only: [:update]
-  before_filter :set_default_cache_headers, only: [:show]
 
   def show
-    item = Rails.application.statsd.time('show.find_by') do
-      ContentItem.find_by(base_path: encoded_base_path)
+    item = Rails.application.statsd.time('show.find_content_item') do
+      ContentItem.where(base_path: encoded_base_path).first
```

Comment:
In Mongoid, a `#find_by` raises an error if the record doesn't exist (annoyingly inconsistent!). I've explicitly raised this error after the cache headers are set further down in this method. The reason to do this was so that I could consolidate the caching behaviour into a single method. Previously it was a bit tangled and scattered throughout the controller.